api/diff.go

```diff
@@ -30,77 +30,72 @@ func apiDiffHandler(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
-type diffResult struct {
-	Diff    shared.ResultsDiff `json:"diff"`
-	Renames map[string]string  `json:"renames,omitempty"`
+func loadDiffRuns(store shared.Datastore, q url.Values) (shared.TestRuns, error) {
+	if runIDs, err := shared.ParseRunIDsParam(q); err != nil {
+		return nil, err
+	} else if len(runIDs) > 0 {
+		runs, err := runIDs.LoadTestRuns(store)
+		// If all errors are NoSuchEntity, we don't treat it as an error.
+		// If err is nil, the type conversion will fail.
+		if multiError, ok := err.(appengine.MultiError); ok {
+			all404s := true
+			for _, err := range multiError {
+				if err != datastore.ErrNoSuchEntity {
+					all404s = false
+					break
+				}
+			}
+			if all404s {
+				return nil, nil
+			}
+		}
+		// err many be non nil here.
+		return runs, err
```

Comment:
Retuning `runs` and `err` together is a bit confusing. What if we add a check on line 51 `else if (err !=nil) {return nil, err)` and return `runs, nil` here?

Reply:
Done.